crypto/params.c

```
@@ -887,3 +891,20 @@ OSSL_PARAM OSSL_PARAM_construct_end(void)
 
     return end;
 }
+
+int OSSL_PARAM_used(const OSSL_PARAM *p)
+{
+    return (p->flags & OSSL_PARAM_USED) != 0;
+}
+
+void OSSL_PARAM_set_unused(OSSL_PARAM *p)
+{
+    if (p != NULL)
+        p->flags &= ~OSSL_PARAM_USED;
+}
+
+void OSSL_PARAM_set_all_unused(OSSL_PARAM *params)
+{
+    while (params != NULL && params->key != NULL)
```

I think silently ignoring a NULL parameter is wrong.